src/utils/api.js

```diff
@@ -0,0 +1,34 @@
+async function fetchApiCall(url, method, body = null) {
+    try {
+      const headers = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'Authorization': 'Token 123456789abcdefg',
+      }
+  
+      const config = {
+        method: method,
+        headers: headers,
+        body: body ? JSON.stringify(body) : null
+      }
+
+      const response = await fetch(url, config)
+  
+      if (!response.ok) {
+        try {
+          const errorData = await response.json()
+          throw new Error(errorData.message || 'Error fetching data')
+        } catch (jsonError) {
+          throw new Error('Error fetching data')
+        }
+      }
+  
+      return await response.json()
+    } catch (error) {
+      console.error('Fetch API Call Error:', error)
+      throw error
+    }
+  }
+  
```

Comment:
I'm having a lot of trouble following the exception handling in this function. Having the entire function wrapped in a try-except block doesn't seem completely necessary. This is what I might do that I think accomplishes the same thing:

```javascript
async function fetchApiCall(url, method, body = null) {
      const headers = {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': 'Token 123456789abcdefg',
      }

      const config = {
        method: method,
        headers: headers,
        body: body ? JSON.stringify(body) : null
      }

      try {
        const response = await fetch(url, config)
      }
      catch(error) {
        console.error("Error raised when sending request.", error)
        throw Error("Error raised when sending request.", error)
      }

      if (!response.ok) {
        console.error("Response not OK", response)
        throw Error("Response not OK", response)
      }
      else {
        return await response.json()
      }
    }
  }
```

This way, we'll always throw a known error (our own) if fetch raises an exception, or the response in not OK (2XX). Then the caller can handle these as needed.

Reply:
By limiting the scope of the try/except blocks we get more granular control of our exceptions and what exception we're passing the caller.

Reply:
thank you for your feedback! I'll make note of this for next time too : )

Reply:
Yes, sorry - I hastily refactored that into something that I thought made sense. Yours looks great

Reply:
sweet, thank you!